server/src/main/java/org/opensearch/indices/NodeIndicesStats.java

```diff
@@ -74,23 +76,22 @@
  */
 @PublicApi(since = "1.0.0")
 public class NodeIndicesStats implements Writeable, ToXContentFragment {
-    private CommonStats stats;
-    private Map<Index, List<IndexShardStats>> statsByShard;
+    protected CommonStats stats;
+    protected Map<Index, CommonStats> statsByIndex;
+    protected Map<Index, List<IndexShardStats>> statsByShard;
 
     public NodeIndicesStats(StreamInput in) throws IOException {
         stats = new CommonStats(in);
+        if (in.getVersion().onOrAfter(Version.V_3_0_0)) {
```

Comment:
By default, we only send and receive `Map<Index, List<IndexShardStats>> statsByShard` even when the level passed on the rest layer may be `indices`. This optimisation we will be sending and receiving `Map<Index, CommonStats> statsByIndex` which is `CommonStats` aggregated on a per Index on a Node. 

This is an optimisation as this was earlier done on the coordinator node side [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/indices/NodeIndicesStats.java#L226). This computation was earlier done by the coordinator node right before building the XContent(JSON)